OutOfSchool/OutOfSchool.AuthCommon/Controllers/AccountController.cs

```diff
@@ -425,6 +429,21 @@ public async Task<IActionResult> ChangePassword(ChangePasswordViewModel model)
         return View("Password/ChangePassword");
     }
 
+    [HttpDelete("{userId}")]
+    [Route("account/deleteuser/{userId}")]
+    [HasPermission(Permissions.ProviderRemove)]
+    public async Task<ResponseDto> LogOutUser(string userId)
+    {
+        if (string.IsNullOrEmpty(userId))
+        {
+            return new ResponseDto() { HttpStatusCode = HttpStatusCode.BadRequest };
+        }
+
+        logger.LogInformation($"LogOut of user with Id = {userId} started");
```

Comment:
(Security vulnerability, logs injection) Sonar is not happy about logging userId. 
Main reason - as `userId` has `string` type it is not processed by framework and thus could contain any data. 
So, please, process it by yourself. If it has to be `Guid` - parse it to this type and use parsed everywhere. If not - at least trim by max length.

Reply:
Thank you, I misunderstood the warning from sonar. I added Guid.TryParse to filter out everything, except Id`s

Reply:
**...and use parsed everywhere...** 
Right now you've parsed it to `id`, but still use unparsed value `userId` in log message

Reply:
`userManager.FindByIdAsync` and `tokenManager.FindBySubjectAsync` take `string` as argument. I used `Guid.TryParse`, only to filter non id strings. Should i use parsed `id` and use `.ToString()` in service?

Reply:
will this work?
`if (string.IsNullOrEmpty(userId) || !Guid.TryParse(userId, out var id))
{
    return new ResponseDto() { HttpStatusCode = HttpStatusCode.BadRequest };
}

logger.LogInformation($"LogOut of user with Id = {id} started");

return await userService.LogOutUserById(id.ToString());`

Or it will be better to change `LogOutUserById` argument from `string` to `Guid`, and use `.ToString()` in `userManager.FindByIdAsync` and `tokenManager.FindBySubjectAsync`

Reply:
Well, it works! At least Sonar no longer shows any vulnerability.

Regarding your other question - should changes of type (`string` -> `Guid`) be promoted deeper in call-stack: ideally yes. Because str**i**ngly-typed coding is not a str**o**ngly-typed. 

In scope of this PR let's stop with current changes. And introduce more improvements in next PR.